core/dataflow/src/main/java/edu/uci/ics/texera/dataflow/source/mysql/MysqlSource.java

```diff
@@ -0,0 +1,161 @@
+    
+    @Override
+    public void open() throws TexeraException {
+        if (cursor == OPENED) {
+            return;
+        }
+        // JDBC connection
+        try {
+            Class.forName("com.mysql.jdbc.Driver").newInstance();
+            String url = "jdbc:mysql://" + predicate.getHost() + ":" + predicate.getPort() + "/"
+                    + predicate.getDatabase() + "?autoReconnect=true&useSSL=true";
+            this.connection = DriverManager.getConnection(url, predicate.getUsername(), predicate.getPassword());
+            
+            statement = connection.createStatement();
+            rs = statement.executeQuery(generateSqlQuery(predicate));
+            cursor = OPENED;
+        } catch (SQLException | InstantiationException | IllegalAccessException | ClassNotFoundException e) {
+            throw new DataflowException("MysqlSink failed to connect to mysql database." + e.getMessage());
+        }
+    }
+    
+    @Override
+    public Tuple getNextTuple() throws TexeraException {
+        if (cursor == CLOSED) {
+            throw new DataflowException(ErrorMessages.OPERATOR_NOT_OPENED);
+        }
+        try {
+			while (rs.next()) {
+				String follower = rs.getString("user.followers_count");
+	    		String friend = rs.getString("user.friends_count");
+	    		follower = follower == null ? "0" :follower;
+	    		friend = friend == null ? "0" :friend;
+	    		String coordinate = rs.getString("coordinate");
+	    		coordinate = coordinate == null ? "" :coordinate;
+	    		String language = rs.getString(TwitterUtils.TwitterSchema.LANGUAGE);
+	    		coordinate = language == null ? "" :language;
+			    Tuple tuple =  new Tuple(this.outputSchema, 
+			            IDField.newRandomID(), //IntegerField
+			            new TextField(rs.getString("text")), 
+			            new StringField(""), new StringField(""), new StringField(""), 
+			            new TextField(rs.getString("user.screen_name")),
+			            new TextField(rs.getString("user.name")),
+			            new TextField(rs.getString("user.description")),
+			            
+			            new IntegerField(Integer.valueOf(follower)),
+			            new IntegerField(Integer.valueOf(friend)),
+			            new TextField(rs.getString("user.location")),
+			            new StringField(rs.getString("user.create_at")),
+			            new TextField(rs.getString("geo_tag.cityName")),
+			            new StringField(coordinate),
+			            new StringField(language)
+			            );
+			    		
+			            //new StringField(resultJsonArray.getJSONObject(cursor).get("ds").toString()));
+			    cursor ++;
+			    
+			    return tuple;
+			}
+		} catch (SQLException e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+		}
+        return null;
+    }
+    
+    public static String generateSqlQuery(MysqlSourcePredicate predicate) {
+    	StringBuilder sb = new StringBuilder();
+    	sb.append("select * from `"+predicate.getTable());
```

Comment:
The MySQL Source operator takes some columns names, instead of doing `select *`, you can select only those columns.